boot/boot_serial/src/boot_serial.c

```diff
@@ -89,7 +89,8 @@ MCUBOOT_LOG_MODULE_DECLARE(mcuboot);
 #endif
 
 static char in_buf[BOOT_SERIAL_INPUT_MAX + 1];
-static char dec_buf[BOOT_SERIAL_INPUT_MAX + 1];
+static char dec_buf[BOOT_SERIAL_INPUT_MAX + 1] __attribute__((aligned));
+static char img_buf[BOOT_SERIAL_INPUT_MAX + 1] __attribute__((aligned));
```

Comment:
I would see this intermediate buffer here only for architectures which requires it. Perfectly this can be added in #ifdef xx #endif. xx should be defined if an arch requires it.

Alternatiovelly flash_area_write() might be added in 3 steps: 1st for unaligned buffer beginning, 2nd for aligned middle buffer part and 3rd for unaligned ending part.

Reply:
At least with the Zephyr build system, values that aren't referenced won't be included in the build. I'm not sure it is worth adding conditional compilation, if the linker is just going to remove the symbols and data anyway.